### ISSUE TITLE
Set policy-bot status to pending if waiting for evaluation result

### ISSUE THREAD

**Issue by KnisterPeter** [palantir/policy-bot#325]
Hi,

I have a feature request so improve the user-experience with the bot.
If there is a rule to wait for a commit status (`has_successful_status`) the bot should set it's status to pending as long as that status to wait for is pending.

This is currently already true for the initial state, but it is not reset to pending.

### Example

The policy bot should wait for a status check with name `ci status`.

> 1. A new PR is created and the status check `ci status` is set to pending by github since it is required.
> The policy-bot status is pending as well now.

> 2. The ci run fails and the `ci status` is set to failure.
> The policy-bot status is set to failure as well now.

> 3. The ci is retriggered and the `ci status` is set to pending again.
> The policy-bot status stays in failure mode.

This last failure state of the policy-bot sets the overall commit/PR status to failure in all list views in github.
I think it would be nice to reset the policy-bot status to pending as long as all statuses to wait for are pending (and no other rule blocks).

**Comment by bluekeyes** [comment]
Thanks for suggesting this. Can you share an example policy that demonstrates the issue? I'd like to experiment with this some to understand the current behavior better before forming an opinion.

Based on your example, I'm guessing that `has_successful_status` appears as a `disapproval` condition. If that is true, I actually expect Policy Bot to post a failure status for (1). Because the pull request does not have the successful status at the time, the overall policy state should be disapproved / failure. The fact that it posts a pending status sounds like a bug (even if the result is desirable.)

**Comment by KnisterPeter** [comment]
Sure I can share that:

```
policy:
  approval:
    - approved by team
approval_rules:
  - name: is maintenance PR
    if:
      has_author_in:
        users: ['some-user']
      has_successful_status:
        - 'ci-status'
    requires:
      count: 1
      users: ['other-user']
    options:
      request_review:
        enabled: true
        mode: 'all-users'
      methods:
        github_review: true
```

This is an excerpt of your configuration with more rules.
My initial goal was to let policy-bot request the review only after the ci was successful. The reason policy-bot sets the PR to failure is that no rules did match. That's totally fine for me, but it doesn't reset the status if the PR is updated or reevaluates the status on some condition.

**Comment by KnisterPeter** [comment]
@bluekeyes Did that help you to understand the feature request?

**Comment by bluekeyes** [comment]
Yes, thank you for sharing the policy and additional details. I didn't realize the failure status was because there were no matching rules. Something still feels wrong about the current behavior (I expect a failure status at step 1 in your example), but I think I'll have to test this out to see how it behaves and how that behavior could be changed. I'm not sure when I'll have time for that.
